experiment/simulation/js/layout.js

```diff
@@ -0,0 +1,161 @@
+import { simulate, deleteElement } from "./gate.js";
+import { connectGate, connectDecoderDisplay, unbindEvent, initDecoder, init7Segment, refreshWorkingArea } from "./main.js";
+import { simulateDecoder } from "./decoder.js";
+"use strict";
+
+// Wires
+export const wireColours = [
+  "#ff0000",
+  "#00ff00",
+  "#0000ff",
+  "#bf6be3",
+  "#ff00ff",
+  "#00ffff",
+  "#ff8000",
+  "#00ff80",
+  "#80ff00",
+  "#ff0080",
+  "#8080ff",
+  "#c0c0c0",
+];
+
+// Contextmenu
+const menu = document.querySelector(".menu");
+const menuOption = document.querySelector(".menu-option");
+let menuVisible = false;
+
+const toggleMenu = command => {
+  menu.style.display = command === "show" ? "block" : "none";
+  menuVisible = !menuVisible;

```

Comment:
Same problem as another experiment here. menuVisible changes separately from the menu's visibility. This may cause inconsistency between them since you are depending on a specific sequence of call to the toggle function. Possibility of subtle bugs

Reply:
fixed